testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/i18n/LoginPageTest.java

```diff
@@ -242,6 +248,31 @@ public void languageUserUpdates() {
         Assert.assertNull(localeCookie);
     }
 
+    // KEYCLOAK-18590
+    @Test
+    public void realmLocalizationMessagesAreNotCachedWithinTheTheme() {
+        final String locale = Locale.ENGLISH.toLanguageTag();
+
+        final String realmLocalizationMessageKey = "loginAccountTitle";
+        final String realmLocalizationMessageValue = "Localization Test";
+
+        CloseableHttpClient httpClient = (CloseableHttpClient) new HttpClientBuilder().build();
```

Comment:
Ensire that you always close any `Response` and `HttpClient` objects, unclosed connections affect stability of the testsuite. Ideally use `try`-with-resources construct to account for the failure as well, see e.g. https://github.com/keycloak/keycloak/blob/c29b8c73a108125511693ec337bc6bb8e9e15cea/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLServletAdapterTest.java#L1365-L1373 for correct sample.
Also see `org.keycloak.testsuite.util.AdminClientUtil.createResteasyClient()` for preferred method to create RestEasy client

Reply:
Hello Hynek, thank you a lot again for the feedback :) .
I put all auto-closables into try-with-resources. The Resteasy client has to be closed manually at the end though.